cmd/compute-domain-controller/computedomain.go

```diff
@@ -63,17 +67,35 @@ type ComputeDomainManager struct {
 
 // NewComputeDomainManager creates a new ComputeDomainManager.
 func NewComputeDomainManager(config *ManagerConfig) *ComputeDomainManager {
-	factory := nvinformers.NewSharedInformerFactory(config.clientsets.Nvidia, informerResyncPeriod)
-	informer := factory.Resource().V1beta1().ComputeDomains().Informer()
+	nvFactory := nvinformers.NewSharedInformerFactory(config.clientsets.Nvidia, informerResyncPeriod)
+	cdInformer := nvFactory.Resource().V1beta1().ComputeDomains().Informer()
+	cdLabelSelector := &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      computeDomainLabelKey,
+				Operator: metav1.LabelSelectorOpExists,
+			},
+		},
+	}
+	cdLabelFactory := informers.NewSharedInformerFactoryWithOptions(
+		config.clientsets.Core,
+		informerResyncPeriod,
+		informers.WithTweakListOptions(func(opts *metav1.ListOptions) {
+			opts.LabelSelector = metav1.FormatLabelSelector(cdLabelSelector)
+		}),
+	)
```

Comment:
I think we should create a `NodeManager` abstraction that is started / stopped by this component rather than implement all of this logic inline here.

Reply:
ack

Reply:
I did that now. Two questions that came up: https://github.com/NVIDIA/k8s-dra-driver-gpu/pull/404#discussion_r2150746190